### DISCUSSION TITLE
SocketTimeoutException on WebService request

### DISCUSSION THREAD

**Discussion by tommyk85** [EmbedITCZ/jbehave-support#560] [Q&A]
Hello,
I'm dealing with a WS call with void response. Moreover I expect to receive read-timeout. How can I go over this step, please?

**Comment by paveljandejsek** [comment]
Void response I guess should be doable with just configuring  getSuccessResult() in web service handler in java code, though not sure about read timeout... You want to skip this step, or just mark timeout as success? Is this one time deal in some specific use case or normal behavior?
From my point of view void and/or timeout sound like an error scenario which should not occur, so I dont expect some easy way to do this - can you elaborate some more about what and why?

**Reply by tommyk85** [reply]
I agree that void response should not be a problem, but the timeout.. The WS request invokes a batch job run and it doesn't provide any response until all steps of the batch job are complete or in case of any failure. Therefore I would need to set longer timeout or considered it as success or just fire and don't care about response.. Further verification can be done by repetitive sql query.

**Reply by tommyk85** [reply]
Actually I expected that `When [ClientRequest] is sent to [MYAPP]` step will not care about the response..

**Reply by paveljandejsek** [reply]
In the end solved with customization in handler:
```
        WebServiceMessageSender[] senders = this.template.getMessageSenders();
        for (WebServiceMessageSender sender : senders) {
            try {
                HttpUrlConnectionMessageSender httpSender = (HttpUrlConnectionMessageSender) sender;
                httpSender.setReadTimeout(Duration.ofMinutes(TIMEOUT_MINS));
                httpSender.setConnectionTimeout(Duration.ofMinutes(TIMEOUT_MINS));
            } catch (ClassCastException ex) {
                log.warn("Cannot set WS timeout: ", ex);
            }
        }
```

@EmbedITCZ/jbehave-support-dev - are we fine with this solution, or should we consider doing explicit support for this in our `WebServiceTemplateConfigurer`?

**Reply by paveljandejsek** [reply]
Resolution: Might be worth it to implement and create issue.
